src/raftstore/store/store.rs

```diff
@@ -2661,3 +2634,99 @@ pub fn new_compaction_listener(ch: SendCh<Msg>) -> CompactionListener {
     };
     CompactionListener::new(compacted_handler, Some(size_change_filter))
 }
+
+fn calc_region_declined_bytes(
+    event: CompactedEvent,
+    region_ranges: &BTreeMap<Key, u64>,
+    bytes_threshold: u64,
+) -> Vec<(u64, u64)> {
+    // Calculate influenced regions.
+    let mut influenced_regions = vec![];
+    for (end_key, region_id) in
+        region_ranges.range((Excluded(event.start_key), Included(event.end_key.clone())))
```

Comment:
why clone the end key but not clone start key here?

Reply:
`end_key` will be used again.